### PR TITLE
Remove legacy classes

### PR DESCRIPTION
**Pull request by rdeutz**
- Removing not needed stuff
- start with composer autoloading
- very basic JLoader to not break B/C
- fixing some Case Sensitive problem

This just a start removing and simplifying code, it keeps the old components running.

### CODE REVIEW

File: administrator/components/com_banners/helpers/html/banner.php [hunk 1]
```diff
@@ -111,6 +111,6 @@ public static function pinned($value, $i, $enabled = true, $checkbox = 'cb')
 			),
 		);
 
-		return JHtml::_('jgrid.state', $states, $value, $i, 'banners.', $enabled, true, $checkbox);
+		return JHtml::_('jGrid.state', $states, $value, $i, 'banners.', $enabled, true, $checkbox);
```
Comment (mbabker) [inline administrator/components/com_banners/helpers/html/banner.php:114]: Short term it'd be easier to just fix the class declaration in `JHtmlJGrid` to be `JHtmlJgrid` (as that matches the expected class name based on JLoader).

File: administrator/index.php [hunk 2]
```diff
@@ -7,41 +7,19 @@
...
-	require_once JPATH_BASE . '/includes/defines.php';
-}
-
-require_once JPATH_BASE . '/includes/framework.php';
-require_once JPATH_BASE . '/includes/helper.php';
-require_once JPATH_BASE . '/includes/toolbar.php';
+/*
+ * Register the auto loader
+ */
+require __DIR__.'/../bootstrap/autoload.php';
```
Comment (mbabker) [inline administrator/index.php:19]: Any way these files can use the existing `includes` folder or we add a new folder in the `libraries` folder?  Adding top level directories isn't always the best thing.

File: index.php [hunk 3]
```diff
@@ -7,39 +7,19 @@
...
-
-/**
  * Constant that is checked in included files to prevent direct access.
- * define() is used in the installation folder rather than "const" to not error for PHP 5.2 and lower
+ * define() is used in the installation folder rather than "const" to
+ * not error for PHP 5.2 and lower
  */
 define('_JEXEC', 1);
 
-if (file_exists(__DIR__ . '/defines.php'))
```
Comment (mbabker) [inline index.php:25]: It looks like we remove the ability to override the app defines, is this intended?  Truthfully we aren't in a state to offer this, so I agree with it, probably best to get opinions from others before committing to it.

File: administrator/index.php (hunk 2 above)
Comment (rdeutz) [inline administrator/index.php:19]: fine by me, though that bootstrap is the name the cool kids are using these days ;-)

File: administrator/components/com_banners/helpers/html/banner.php (hunk 1 above)
Comment (rdeutz) [inline administrator/components/com_banners/helpers/html/banner.php:114]: I fixed it on all occurrences, we have the same problem for contentLanguages sometime with and sometimes without uppercase "L". So we might need to make a general decision what we will do with class names. If we are going to use the composer autoloader it is needed that the names match exactly.

Comment (mbabker) [inline administrator/components/com_banners/helpers/html/banner.php:114]: If we use the existing rules for JLoader where every capital letter is supposed to represent a new directory then that should let us deal with inconsistencies like with `JHtmlJGrid` or `JHtmlContentLanguage` which are wrong.

**Comment by nibra** [comment]
Can't be moved to `joomla/joomla-cms` without a 4.0 branch.

**Pull request closed without merging**